src/Orc.FileAssociation.Tests/FileAssociationsServiceFacts.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Orc.FileAssociation.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class FileAssociationsServiceFacts
+    {
+        [Test]
+        [TestCaseSource(nameof(Cases))]
+        public async Task AssociateFilesWithApplicationAsyncTestAsync(ApplicationInfo applicationInfo, List<string> extensions, string path)
+        {
+            var fileAssociationServiceMock = new FileAssociationServiceMock();
+            await fileAssociationServiceMock.AssociateFilesWithApplicationAsync(applicationInfo);
+
+            CollectionAssert.AreEqual(extensions, fileAssociationServiceMock.Extensions);
```

Comment:
Shouldn't we create the actual registry keys here and test (using win32 registry calls) whether they are correctly added and removed? If we create a mock like this, what are we really testing?